app/src/main/java/bg/zahov/app/ui/settings/profile/EditProfileViewModel.kt

```diff
@@ -1,121 +1,160 @@
+     *
+     * @property authenticated Whether the user is authenticated.
+     * @property username The current username.
+     * @property password The current password.
+     * @property passwordVisibility Controls the visibility of the password.
+     * @property notify Optional notification message.
+     */
+    data class UiState(
+        val authenticated: Boolean = false,
+        val username: String = "",
+        val password: String = "",
+        val passwordVisibility: Boolean = false,
+        val notify: String? = null,
+    )
 
-    private val _name = MutableLiveData<String>()
-    val name: LiveData<String>
-        get() = _name
+    private val _state = MutableStateFlow(UiState())
+    val state: StateFlow<UiState> = _state
 
+    /**
+     * Updates the username in the UI state.
+     *
+     * @param username The new username to be set.
+     */
+    fun onUsernameChange(username: String) {
+        _state.update { it.copy(username = username) }
+    }
 
-    private val _isUnlocked = MutableLiveData(false)
-    val isUnlocked: LiveData<Boolean>
-        get() = _isUnlocked
+    /**
+     * Updates the password in the UI state.
+     *
+     * @param password The new password to be set.
+     */
+    fun onPasswordChange(password: String) {
+        _state.update { it.copy(password = password) }
+    }
 
+    /**
+     * Toggles the visibility of the password field.
+     */
+    fun onPasswordVisibilityChange() {
+        _state.update { it.copy(passwordVisibility = !_state.value.passwordVisibility) }
+    }
 
     init {
         viewModelScope.launch {
             try {
                 repo.getUser().collect {
-                    _name.postValue(it.name)
+                    _state.update { old -> old.copy(username = it.name) }
                 }
             } catch (e: CriticalDataNullException) {
-                serviceErrorHandler.initiateCountdown()
+                errorHandler.initiateCountdown()
             }
         }
     }
 
-    fun updateUsername(newUsername: String) {
-        if (newUsername.isEmpty()) {
-            _state.value = State.Notify("Username cannot be empty")
-            return
-        }
-
+    /**
+     * Updates the username in the repository and sets a notification message upon success or failure.
+     */
+    fun updateUsername() {
+        val newUsername = _state.value.username
         viewModelScope.launch {
             repo.changeUserName(newUsername)
                 .addOnSuccessListener {
-                    _name.postValue(newUsername)
-                    _state.postValue(State.Notify("Successfully updated username"))
-                }
-                .addOnFailureListener {
-                    _state.postValue(State.Notify("Successfully updated username"))
+                    _state.update { old ->
+                        old.copy(
+                            username = newUsername,
+                            notify = "Successfully updated username"
```

Comment:
Haven't we used string reseources in previous message handling pr-s?

Reply:
I dont think so. Will change it :)

Reply:
if we didn't use string resources, create a toast manager handler task and leave it, if you think the solution it's not straightforward when you are using string resources

Reply:
you can see this approach https://github.com/android/compose-samples/blob/main/Jetsnack/app/src/main/java/com/example/jetsnack/model/SnackbarManager.kt